src/Amazon.CloudWatch.EMF/Sink/AgentSink.cs

```diff
@@ -118,7 +118,7 @@ private Task RunSenderThread(ILoggerFactory loggerFactory)
                             }
                             catch (Exception e)
                             {
-                                logger.LogWarning("Failed to write message to socket. Backing off and trying again. {}", e.Message);
+                                logger.LogWarning(e, "Failed to write message to socket. Backing off and trying again.");
                                 Thread.Sleep(1000); // TODO: backoff
```

Comment:
I'm not sure this shouldn't be:

```csharp
await Task.Delay(TimeSpan.FromMilliseconds(1000)).ConfigureAwait(false);
```

…but that felt out of scope and like a Chesterton's Fence thing.